File: Softeer/스마트 물류.py
from sys import stdin
n, k = map(int, stdin.readline().split())
line = list(map(str, stdin.readline().rstrip()))
# P 로봇, H 부품

res = 0
visited = [0] * n
for i in range(n):
    if line[i] == 'P':
        for val in range(i - k, i + k + 1):
            if n > val >= 0:
                if line[val] == 'H' and not visited[val]:
                    res += 1
                    visited[val] = 1
                    break
print(res)  

File: Softeer/스마트 물류.js
var input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
var [n, k] = input[0].split(' ').map(Number);
var lst = input[1].split('');
visited = new Array(n).fill(0);

var res = 0;
for(let i = 0; i < n; i++){
    if(lst[i] === 'P'){
        for(let j = i-k;j<i+k+1;j++){
            if(0<=j && j < n){
                if(lst[j] === 'H' && visited[j] === 0){
                    visited[j] = 1;
                    res += 1;
                    break;
                }
            }
        }
    }
}
console.log(res)